python-lib/text_extraction/__init__.py

```diff
@@ -131,8 +133,8 @@ def extract_markdown_chunks(markdown, filename):
 
         # Check each line against each of the header types (e.g., #, ##), header_level is the number of '#'
         for header_level in range(1, 4):
-            # Check if line starts with a header that we intend to split on
-            if stripped_line.startswith("#" * header_level) and (
+            # Check if line starts with a header (tab/4-spaces means code block) that we intend to split on
+            if stripped_line.startswith("#" * header_level) and not stripped_line.startswith(("    ", "\t")) and (
```

Comment:
There is something weird with this condition.

- If the row is `# header` then the second part of the `and` will always be true
- If the row is `    # some code comment`, then first part will always be false and we will never check the second part of the condition

So I guess adding `not stripped_line.startswith(("    ", "\t"))` does change the behavior here.

Am I missing something ?

Reply:
fixed in [2902d99](https://github.com/dataiku/dss-plugin-tesseract-ocr/pull/73/commits/2902d994261aea9687828d9fdd7160ba37632d34)